packages/discovery-react-components/src/components/DocumentPreview/DocumentPreview.tsx

```diff
@@ -78,36 +89,14 @@ const DocumentPreview: FC<Props> = ({
     // setLoading(true);
   }, [doc]);
 
-  // fetch PDF (if necessary)
-  useEffect(() => {
-    async function fetchFile() {
-      const hasFile = await documentProvider!.provides(document!);
-      if (hasFile) {
-        const file = await documentProvider?.get(document!);
-        if (typeof file === 'string' || file instanceof ArrayBuffer) {
-          setProvidedFile(file);
-        } else if (file?.type === 'pdf') {
-          setProvidedFile(file.source);
-        } else {
-          setProvidedFile(undefined);
-        }
-      }
-    }
-
-    // `file` takes precedence over a file provided by `documentProvider`
-    if (file) {
-      setProvidedFile(file);
-    } else if (document && documentProvider) {
-      fetchFile();
-    } else {
-      setProvidedFile(undefined);
-    }
-  }, [document, documentProvider, file]);
-
   const [pdfPageCount, setPdfPageCount] = useState(0);
 
   const base = `${settings.prefix}--document-preview`;
 
+  if (fetching) {
+    return <Loading withOverlay={false} />;
```

Comment:
nit: it'd nice if we can apply a `className` so that we can customize the style.